core/domain/html_validation_service.py

```diff
@@ -875,3 +875,17 @@ def get_filename_with_dimensions(old_filename, exp_id):
     new_filename = regenerate_image_filename_using_dimensions(
         old_filename, height, width)
     return new_filename
+
+
+def get_invalid_svg_tags(svg_string):
+    """Returns a set of all invalid tags for the provided SVG.
+
+    Args:
+        svg_string: str. The SVG string.
+
+    Returns:
+        set. A set of invalid tags.
```

Comment:
Shouldn't we be whitelisting attributes on tags as well?

Reply:
Per discussion on email with @BenHenning, it was decided to introduce whitelisting of "fully supported" tags mentioned [here](http://bigbadaboom.github.io/androidsvg/). I think that means all attributes for those tags would be allowed. Would that work or do we need to explicitly specify which attributes are allowed? If we decide to do the latter, I think we could use the same list of allowed SVG attributes specified in the AngularJS sourcecode [here](https://github.com/angular/angular.js/blob/418355f1cf9a9a9827ae81d257966e6acfb5623a/src/ngSanitize/sanitize.js#L384).